flake.nix

```diff
@@ -0,0 +1,45 @@
+{
+  description = "Codex build flake";
+  
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.05";
+  };
+
+  outputs = { self, nixpkgs }:
+    let
+      supportedSystems = [
+        "x86_64-linux" "aarch64-linux"
+        "x86_64-darwin" "aarch64-darwin"
+      ];
+      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
+      pkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
+    in rec {
+      devShells = forAllSystems (system: let
+        pkgs = pkgsFor.${system};
+        inherit (pkgs) lib stdenv mkShell;
+      in {
+        default = mkShell.override { stdenv = pkgs.gcc11Stdenv; } {
+          buildInputs = with pkgs; [
+              # General
+              git pkg-config openssl lsb-release
+              # Build
+              rustc cargo nim nimble gcc11 cmake
+              # Libraries
+              gmp llvmPackages.openmp
+              # Tests
+              nodejs_18
+          ];
+          shellHook = ''
+              export NIMBLE_DIR=$HOME/.nimble
+              export RUSTFLAGS="-C target-cpu=native"
+              export CXXFLAGS="-std=c++17 -march=native -mtune=native -msse -msse2 -msse3 -mssse3 -msse4.1 -msse4.2 -mavx -mavx2 -fopenmp"
+              export CFLAGS="-march=native -mtune=native -msse -msse2 -msse3 -mssse3 -msse4.1 -msse4.2 -mavx -mavx2"
```

Comment:
Why do we need these? Were they recommended by devs? If it is it probably should be set in the `Makefile`.

Reply:
I didn't ask(I will) but without them build would fail

Reply:
That's weird. If it's required for builds it should be in `Makefile`. It would be good to know what part of this is actually necessary. I'd rather avoid [cargo culting](https://en.wikipedia.org/wiki/Cargo_cult) stuff into config files that are already hard to follow for most people.

Reply:
I've moved the CXXFLAGS env variable to makefile with **minimum** number of compiler flags in order for build to work. I'll add someone from codex team so they can see this change and comment on it.

Reply:
I never had to set anything of the kind to build Codex, I wonder why this is needed here.